core/server/models/base/plugins/crud.js

```diff
@@ -179,6 +183,8 @@ module.exports = function (Bookshelf) {
                 model.hasTimestamps = false;
             }
 
+            options.lock = 'forUpdate';
```

Comment:
Based on the doc from Bookshelf:
> Type of row-level lock to use. Valid options are forShare and forUpdate. This only works in conjunction with the transacting option, and requires a database that supports it.

I think it's worth adding an explicit check for 'transacting' here. Otherwise we might give an impression that the operation is always protected by the lock.

We do have similar check in our collections. I'm assuming the thinking was similar:
```js
        onFetchingCollection: function onFetchingCollection(model, columns, options) {
            if (options.forUpdate && options.transacting) {
                options.query.forUpdate();
            }
        },
```